### PR TITLE
feat: Require `Immutable` for `Channel.send`

### LINKED ISSUE #4846: Require `Imm` instance for `Channel.send` and `Channel.recv`
- [ ] Modify the signature of `Channel.send` to require an `Imm[t]` instance.
- [x] ~Modify the signature of `Channel.recv` to require an `Imm[t]` instance.~

This ensures that channels can only be used for immutable message passing; we cannot share mutable data, hence we cannot have data races. (This is strictly not true yet, because a spawn expression could close over a shared mutable data structure-- this we have to resolve).

### PR DESCRIPTION
**Pull request by paulbutcher**
Fixes #4846

### CODE REVIEW

File: main/src/library/Boxed.flix [hunk 1]
```diff
@@ -32,6 +32,8 @@ pub enum Boxed {
     )
 }
 
+instance Immutable[Boxed]
```
Comment (magnus-madsen) [inline main/src/library/Boxed.flix:35]: Any reason not to use `with Immutable`?

File: main/src/library/Channel.flix [hunk 2]
```diff
@@ -41,7 +41,17 @@ namespace Channel {
...
     ///
-    pub def send(m: t, s: Sender[t]): Unit \ IO = $CHANNEL_PUT$(s, m)
+    pub def send(m: t, s: Sender[t]): Unit \ IO with Immutable[t] = $CHANNEL_PUT$(s, m)
+
+    ///
+    /// Sends the message `m` on the given channel `s`.
+    ///
+    /// Blocks until the message is enqueued.
+    ///
+    /// Identical to `send` but doesn't require `Immutable`
```
Comment (magnus-madsen) [inline main/src/library/Channel.flix:51]: . 
It is up to programmer to ensure that race conditions cannot arise due to concurrent access to `m`.

File: main/src/library/Boxed.flix (hunk 1 above)
Comment (paulbutcher) [reply main/src/library/Boxed.flix:35]: I guess I was just aiming to be consistent with the surrounding code, but I'm happy to switch to `with` if you prefer. Let me know.

Comment (magnus-madsen) [reply main/src/library/Boxed.flix:35]: I would propose we use maximal derivation, i.e. place it on the enum, when possible.

**Pull request merged**